api/envoy/extensions/filters/http/alternate_protocols_cache/v3/alternate_protocols_cache.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.alternate_protocols_cache.v3;
+
+import "envoy/config/core/v3/protocol.proto";
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.alternate_protocols_cache.v3";
+option java_outer_classname = "AlternateProtocolsCacheProto";
+option java_multiple_files = true;
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Alternate Protocols Cache]
+
+// Configuration for the alternate protocols cache HTTP filter.
+// [#extension: envoy.filters.http.alternate_protocols_cache]
+// TODO(RyanTheOptimist): Move content from source/docs/http3_upstream.md to
+// docs/root/intro/arch_overview/upstream/connection_pooling.rst when unhiding the proto.
+message FilterConfig {
+  // [#not-implemented-hide:]
+  // The presence of alternate protocols cache options causes the use of the
+  // alternate protocols cache, which is responsible for parsing and caching
+  // HTTP Alt-Svc headers. This enables the use of HTTP/3 for origins that
```

Comment:
Does `origins` mean upstream servers?

Also, can you clarify what happens if HTTP/3 is not listed as part of a cluster's protocol options, but the configuration includes this filter config? (from reading the comment it seems that HTTP/3 and alternate protocols cache will still be used if the upstream supports it).